src/App.js

```diff
@@ -1,24 +1,52 @@
-import React from 'react';
+import React, {useState} from 'react';
 import logo from './logo.svg';
 import './App.css';
+import Contacts from './Contacts';
+import importedContacts from './contacts.json';
+
+const contactsArray = importedContacts.splice(0, 5);
 
 function App() {
+
+  const [contacts, setContacts] = useState(contactsArray);
+
+  const addRandomContact = () => {
+    let copyContacts = [...contacts];
+    let randomContact = '';
+    
+    do {
+      randomContact = importedContacts[Math.floor(Math.random() * importedContacts.length)];
+    } while (contacts.some(actor => actor.id === randomContact.id));
```

Comment:
Instead of this loop, you could simply splice the selected random contact from the importedContacts, and push that contact back again to the importedContacts upon deletion.

Reply:
but then how do I know that the random contact is not already in the contacts that are in the array?

Reply:
If you splice contact X from the importedContacts, and push it to the contacts in the state, there is no X in importedContacts, and X in the state contacts. Upon deletion, contact X can be spliced from the state contacts, and be pushed to the importedContacts.